develop/learning-paths/migration/articles/01-intro/01-from-a-monolith-to-a-modular-architecture.markdown

```diff
@@ -0,0 +1,124 @@
+even better capabilities while making Liferay Portal, and its features and
+frameworks, lean, agile, and independent. To start, we took a hard look at key
+capabilities to improve.
+
+Liferay Portal's releasability was something we definitely wanted to improve.
+Before Liferay 7, you couldn't download releases of new features to *parts* of
+Liferay Portal--you had to wait until a new release of the *entire* portal
+product. And you couldn't just take the new parts you wanted, you had to take
+all of Liferay Portal--the whole monolith. Releasability of Liferay Portal's
+core and its components became a top priority.
+
+To improve releasability, we came face to face with how Portal's parts were
+interacting. In many cases, they had too much knowledge of each other's
+internals (i.e., they were coupled tightly). Tight coupling was occurring way too
+easily in the development process, as we hadn't yet found an easy way to facilitate and enforce
+loose coupling. So we also prioritized finding a framework to help Liferay
+engineers and contributors develop loosely coupled components on Liferay.
+
+One last major goal related to both releasability and loose coupling was finding
+the best way to modularize Liferay's core, apps, and frameworks into independent
+lightweight components. We wanted to do away with the massive monolith and
+refactor it into components that would be easy to maintain and iterate on
+separately and frequently. In our research, we drew inspiration from the concept
+of microservices, which we'll discuss next. 
+
+## The Microservices Concept
+
+The goal of a microservices is to do one (and only one) thing very
+well. Each microservice is small and stripped down of all but what it requires
+to meet its purpose. Their independent too. You can independently deploy each
+microservice and use it. But in addition to being independent, each microservice
+is composable, meaning it can be used with other microservices to do more
+complex things. 
+
+![Figure 1: Liferay drew inspiration from the concept of microservices. Their size, self reliance, and composability, make them ideal for a modular architecture.](../../images/microservice.png)
+
+In essence, we've rebuilt Liferay as microservices, to gain the ability to
+deploy new versions of the services in a few weeks, or even days, as compared
+previously to months. We've encapsulated these microservices in OSGi modules,
+which are a standards based deployment structure for deploying to dynamic Java
+systems.
+
+All of Liferay 7's modules can be developed and iterated on independently. Each
+module can have it's own release cycle. Developers can innovate rapidly over
+their modules. As with their modules, they can focus on providing the single
+purpose of their module.
+
+We'll go into details of OSGi modules later. For now, take comfort that we've
+chipped away at the Liferay 6.2 monolith to bring you Liferay 7 as a robust
```

Comment:
Exactly what we *don't* want to do: chip away at 6.2. We want 6.2 to stand tall and high on its own, and we also want to say that 7.0 is better.